src/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -141,7 +141,21 @@ private string ReadLine(bool consumeKeys)
                                     string moveLeft = ConsolePal.TerminalFormatStrings.Instance.CursorLeft;
                                     s_moveLeftString = !string.IsNullOrEmpty(moveLeft) ? moveLeft + " " + moveLeft : string.Empty;
                                 }
-                                Console.Write(s_moveLeftString);
+
+                                // The ReadLine input may wrap accross terminal rows and we need to handle that.
+                                // note: ConsolePal will cache the cursor position to avoid making many slow cursor position fetch operations.
+                                if (ConsolePal.TryGetCursorPosition(out int left, out int top, reinitializeForRead: true) &&
+                                    left == 0 && top > 0)
+                                {
+                                    // Move to end of previous line
+                                    ConsolePal.SetCursorPosition(ConsolePal.WindowWidth - 1, top - 1);
```

Comment:
What happens if the window is resized?

Reply:
It behaves as expected when I try this using gnome terminal and terminator. @laceous [reported](https://github.com/dotnet/corefx/issues/37056#issuecomment-488530388) it is not behaving ok with Konsole. I'm not sure if we should try to fix that. If we do, only the active ReadLine will be properly wrapped according to the new window size, anything written before that will still be responsibility of the terminal emulator.